src/App.tsx

```diff
@@ -1,13 +1,27 @@
-import React, { ReactNode } from "react";
+import React, { useState } from "react";
+import './styles/style.scss';
+import Feedback from "./components/Feedback/Feedback";
+import Market from './components/Market/Market';
+import Navigation from './components/Navigation';
 
-const App = () : ReactNode => {
+const componentsMap = {
+  feedback: Feedback,
+  market: Market,
+};
 
-    return (
-        <>
-          <h1>Widget</h1>
-        </>
-      );
-      
+const App = (): JSX.Element => {
+  const [currentModal, setCurrentModal] = useState('feedback');
+  const CurrentComponent = componentsMap[currentModal];
+
+  return (
+    <>
+      <Navigation 
+        currentModal = {currentModal}
+        setCurrentModal={setCurrentModal} 
+      />
+      {CurrentComponent && <CurrentComponent />}
```

Comment:
```
import React, { useState } from "react";
import 'shared/styles/layout/layout.scss';
import Feedback from "../widgets/feedback/feedback/Feedback";
import Market from '../widgets/market/market/Market';
import Navigation from '../widgets/navigation/Navigation';

export type ModalKeys = 'feedback' | 'market' | 'settings' | 'subscribe'; 

const componentsMap = {
  feedback: Feedback,
  market: Market,
  settings: Feedback,
  subscribe: Market
};

const App = (): JSX.Element => {
  const [currentModal, setCurrentModal] = useState<ModalKeys>('feedback');

  const CurrentComponent = componentsMap[currentModal];

  //если вместо CurrentComponent возвращать {componentsMap[currentModal]}:
  // Functions are not valid as a React child. 
  //This may happen if you return a Component instead of <Component /> from render. 

  return (
    <>
      <Navigation 
        currentModal={currentModal}
        setCurrentModal={setCurrentModal} 
      />
      <CurrentComponent/>
    </>
  )
}

export default App;
```
проверку убрала, типизировала, сделала union для типа, но ршибка если {componentsMap[currentModal]} ставлю

Reply:
Прочитай внимательно текст ошибки и исправь.